torchdynamo/convert_frame.py

```diff
@@ -101,6 +103,27 @@ def _fn(*args, **kwargs):
     return _fn
 
 
+def has_tensor_in_frame(frame):
+    # Check if the passed arguments are of type Tensor
+    for value in frame.f_locals.values():
+        if isinstance(value, torch.Tensor):
+            return True
+
+    # Check if there is global import of torch.*
+    for value in frame.f_globals.values():
```

Comment:
It might be faster to check only `for name in frame.f_code.co_names`.  This will give only the globals used by the function.